dashboard/cron.py

```diff
@@ -23,100 +23,129 @@
 
 logger = logging.getLogger(__name__)
 
-engine = db_util.create_sqlalchemy_engine(settings.DATABASES['default'])
-data_warehouse_engine = db_util.create_sqlalchemy_engine(settings.DATABASES['DATA_WAREHOUSE'])
+# cron job to populate course and user tables
+class DashboardCronJob(CronJobBase):
 
-# Set up queries array from configuration file
-CRON_QUERY_FILE = settings.CRON_QUERY_FILE
-logger.info(CRON_QUERY_FILE)
-try:
-    with open(CRON_QUERY_FILE) as cron_query_file:
-        queries = hjson.load(cron_query_file)
-except FileNotFoundError:
-    logger.error(
-        f'Cannot find cron queries file "{CRON_QUERY_FILE}".')
-    queries = dict()
+    schedule = Schedule(run_at_times=settings.RUN_AT_TIMES)
+    code = 'dashboard.DashboardCronJob'    # a unique code
 
-# Split a list into *size* shorter pieces
+    def setup_queries(self):
+        # Set up queries array from configuration file
+        CRON_QUERY_FILE = settings.CRON_QUERY_FILE
+        logger.info(CRON_QUERY_FILE)
+        try:
+            with open(CRON_QUERY_FILE) as cron_query_file:
+                self.queries = hjson.load(cron_query_file)
+        except FileNotFoundError:
+            logger.error(f'Cannot find cron queries file "{CRON_QUERY_FILE}".')
 
+    def setup_bigquery(self):
+        # Instantiates a client
+        self.bigquery_client = bigquery.Client()
+#        self.bigquery_client = bigquery.Client.from_service_account_json(settings.BIGQUERY_SERVICE_ACCOUNT_JSON)
```

Comment:
I remember that I had this in there because I was thinking of switching the way we configure BigQuery to use an explicit JSON value in the hjson rather than the bq_cred.json file. This is probably a separate issue but something I noticed in the newer version of the library that is supported.